storage/series/series.go

```diff
@@ -603,25 +604,19 @@ func (s *dbSeries) Flush(
 	blockStart time.Time,
 	persistFn persist.Fn,
 ) (FlushOutcome, error) {
-	// NB(r): Do not use defer here as we need to make sure the
-	// call to sr.Segment() which may fetch data from disk is not
-	// blocking the series lock.
 	s.RLock()
+	defer s.RUnlock()
```

Comment:
is the disk fetch (earlier comment) no longer a concern?

Reply:
I think he removed it because it only applied to cache all metadata and we're planning on deleting that

Reply:
Yeah we're going to delete the cache all metadata strategy, also it's probably not all that bad to just wait if that's the case (it never should though though as this is a flush, the data should never have come from disk if we're flushing this).

Reply:
fair enough